File: frontend/src/types/telegram.ts
export interface TelegramSubscriber {
  id: number;
  created_at: string;
  updated_at: string;
  chat_id: string;
  username: string | null;
  is_active: boolean;
  notify_trade_opened: boolean;
  notify_trade_closed: boolean;
  notify_sl_tp_hit: boolean;
  notify_high_impact_news: boolean;
  notify_daily_summary: boolean;
  notify_new_setup: boolean;
}

export interface TelegramSubscriberCreate {
  chat_id: string;
  username?: string | null;
}

export interface TelegramPreferencesUpdate {
  is_active?: boolean;
  notify_trade_opened?: boolean;
  notify_trade_closed?: boolean;
  notify_sl_tp_hit?: boolean;
  notify_high_impact_news?: boolean;
  notify_daily_summary?: boolean;
  notify_new_setup?: boolean;
}

export interface TelegramTestMessageRequest {
  chat_id?: string | null;
  message: string;
}

File: backend/app/schemas/telegram.py
"""Pydantic schemas for Telegram subscriber management."""

from __future__ import annotations

from pydantic import BaseModel

from app.schemas.common import TimestampedSchema


class TelegramSubscriberCreate(BaseModel):
    chat_id: str
    username: str | None = None


class TelegramPreferencesUpdate(BaseModel):
    is_active: bool | None = None
    notify_trade_opened: bool | None = None
    notify_trade_closed: bool | None = None
    notify_sl_tp_hit: bool | None = None
    notify_high_impact_news: bool | None = None
    notify_daily_summary: bool | None = None
    notify_new_setup: bool | None = None


class TelegramSubscriberRead(TimestampedSchema):
    chat_id: str
    username: str | None = None
    is_active: bool
    notify_trade_opened: bool
    notify_trade_closed: bool
    notify_sl_tp_hit: bool
    notify_high_impact_news: bool
    notify_daily_summary: bool
    notify_new_setup: bool


class TelegramTestMessageRequest(BaseModel):
    chat_id: str | None = None
    message: str = "✅ Fathir AI Trading Assistant is connected."
